apps/Minecraft Pi (Modded)/install

```diff
@@ -1,37 +1,66 @@
+fi
+
+# Get OS info for later
+source /etc/os-release
+RELEASE="$(awk '{print int($1)}' <<<$VERSION_ID)"
+
 # Remove old MCPILs
 sudo apt-get remove -y mcpil-r mcpil &>/dev/null || true
 
+# Install mcpi-packages apt repo
 wget -qO- https://raw.githubusercontent.com/MCPI-Revival/mcpi-packages/master/install.sh | grep -v 'sudo apt update' | sudo bash - || error "Failed to install APT repo!"
 
-# Choose MCPIL to use
-PS3="Which launcher for MCPI Modded would you like to use? Need help deciding? Ask here: https://discord.com/invite/aDqejQGMMy
+# Choose launcher to use
+PS3="Which launcher for MCPI Modded would you like to use? Need help deciding? Ask here: https://discord.com/invite/aDqejQGMMy. You can also go to the github links for the respective launchers to see how they look.
 "
-options=("gMCPIL (recommended)" "jMCPIL (better GUI)" "Quit")
+options=("jMCPIL (recommended, better looking GUI but requires Java) (https://github.com/MCPI-Revival/jMCPIL)" "gMCPIL (written in Vala and C) (https://github.com/MCPI-Revival/gMCPIL)")
 select opt in "${options[@]}"
 do
   case $opt in
     "${options[0]}")
-        MCPIL="gmcpil"
+        MCPIL="jmcpil"
         break;;
     "${options[1]}")
-        MCPIL="jmcpil"
+        MCPIL="gmcpil"
         break;;
-    "${options[2]}")
-        exit 0;;
     *) echo "Invalid option: $REPLY";;
   esac
 done
 
-install_packages $MCPIL python3-pip python3-dev || error "Failed to install dependencies"
-pip3 install mcpi || error "The pip3 command failed to install mcpi!"
+# Install the launcher
+"${DIRECTORY}/pkg-install" "$MCPIL" "$(dirname "$0")" || error "Failed to install ${MCPIL} package!"
+
+# Optionally install sound if on Bullseye
+if [[ "$ID" == "debian" || "$ID" == "raspbian" || "$ID" == "ubuntu" ]]; then
+	[[ "$RELEASE" -lt 11 ]] && echo "OS version smaller than bullseye, not displaying sound prompt" && break
+	read -p "Would you like to download sound files so MCPE sound effects play in-game? (e.g walking effects, placing blocks, etc)" choice
+  case "$choice" in 
+  y|Y ) wget -o libminecraftpe.so https://archive.org/download/libminecraftpe0.6.1/libminecraftpe06%2B08.so && mkdir -p ~/.minecraft-pi/overrides && mv libminecraftpe.so ~/.minecraft-pi/overrides/libminecraftpe.so || error "Failed to install sound files!";;
+  n|N ) echo "Not installing sound files... proceeding";;
+  * ) echo "Invalid option! Please type 'y' or 'n'";;
+esac
+fi
+
+# Optionally install Python API
+read -p "Do you wish to install the MCPI API for the ability to run Python API scripts? (select yes unless you know what your doing!)" choice
+case "$choice" in 
+  y|Y ) "${DIRECTORY}/pkg-install" "python3-pip" "$(dirname "$0")" || error "Failed to install python3-pip which is needed to install the API!" && pip3 install mcpi || error "The pip3 command failed to install the MCPI API!";;
+  n|N ) echo "Not installing API... proceeding";;
+  * ) echo "Invalid option! Please type 'y' or 'n'";;
+esac
```

Comment:
Put this whole read and case inside a loop so if user enters an invalid answer, they can enter an answer again.

Reply:
> 

Like a forever loop and use `break`?

Reply:
yes